### PR TITLE
PB-1043: Update the pre-created user when it signs up via special link on the email

### PR DESCRIPTION
**Pull request by edisonjpadilla**

### CODE REVIEW

File: src/modules/user/user.controller.ts [hunk 1]
```diff
@@ -150,6 +153,108 @@ export class UserController {
...
+      await this.userUseCases.replacePreCreatedUser(
+        userCreated.user.email,
+        userCreated.user.uuid,
+        keys.publicKey,
+      );
+
+      await this.sharingService.acceptInvite(userCreated.user, invitationId, {
+        encryptionKey: createUserDto.publicKey,
+        //TODO: What properties should be here?
+        encryptionAlgorithm: '',
```
Comment (edisonjpadilla) [inline src/modules/user/user.controller.ts:202]: @sg-gs  @apsantiso please tell me what property should be here?

Comment (apsantiso) [reply src/modules/user/user.controller.ts:202]: The invite has the encryption key and algorithm in this case, dont send them. The user is not inviting himself to the sharing.

`    const newSharing = Sharing.build({
      ...invite,
      id: v4(),
      type: SharingType.Private,
      ownerId: owner.uuid,
      encryptionAlgorithm: invite.isARequest()
        ? acceptInviteDto.encryptionAlgorithm
        : invite.encryptionAlgorithm,
      encryptionKey: invite.isARequest()
        ? acceptInviteDto.encryptionKey
        : invite.encryptionKey,
    });`

Comment (edisonjpadilla) [reply src/modules/user/user.controller.ts:202]: perfect, thanks!

**Review by apsantiso — CHANGES_REQUESTED** [review]

File: src/modules/user/user.controller.ts [hunk 2]
```diff
@@ -150,6 +153,109 @@ export class UserController {
...
+    @Req() req: Request,
+    @Res({ passthrough: true }) res: Response,
+  ) {
+    try {
+      const preCreatedUser = await this.userUseCases.findPreCreatedByEmail(
+        createUserDto.email,
+      );
+
+      if (!preCreatedUser) {
+        throw new NotFoundException('PRE_CREATED_USER_NOT_FOUND');
```
[the remark this answers is not in the document]
Comment (edisonjpadilla) [reply src/modules/user/user.controller.ts:182]: That's a message to the front, the front gets that message and shows whatever they want to the user

File: src/modules/user/user.controller.ts [hunk 3]
```diff
@@ -150,6 +153,108 @@ export class UserController {
...
   }
 
+  @UseGuards(ThrottlerGuard)
+  @Throttle({
+    long: {
+      ttl: 3600,
+      limit: 5,
+    },
+  })
+  @Post('/')
```
[the remark this answers is not in the document]
Comment (edisonjpadilla) [reply src/modules/user/user.controller.ts:163]: yes. you right

**Review by apsantiso — APPROVED** [review]

**Pull request merged**